pwiz_tools/Shared/zedgraph/ZedGraph/LabelLayout.cs

```diff
@@ -319,10 +326,13 @@ public bool PlaceLabel(LabeledPoint labPoint, Graphics g)
             var roughGoal = goal;
             // Search the cell neighborhood for a better position
             var goalPoint = _densityGrid[goalCell.Y][goalCell.X]._location;
-            for (var count = 15; count > 0; count--)
+            var chartRect = _graph.Chart.Rect;
+            var allowedRect = new RectangleF(chartRect.X + labelRect.Width / 2, chartRect.Y,
```

Comment:
I am not sure I understand why `labelRect.Width / 2` gets added to the X-coordinate, but a similar sort of thing does not get added to the Y-coordinate.
This code might be correct, but I wanted to point out that it looks suspicious.

Reply:
This is because label coordinates are top-middle, so Y coordinate is not adjusted.